cmd/external-repos/main.go

```diff
@@ -158,16 +154,64 @@ func saveToDB(ctx context.Context, db *gorm.DB) error {
 		return err
 	}
 	urls = external_repos.GetBaseURLs(extRepos)
-	err = dao.RepositoryConfig.SavePublicRepos(ctx, urls)
+	err = daoReg.RepositoryConfig.SavePublicRepos(ctx, urls)
 	if err != nil {
 		return err
 	}
 
-	rh := external_repos.NewRedHatRepos(dao)
+	rh := external_repos.NewRedHatRepos(daoReg)
 	err = rh.LoadAndSave(ctx)
+	if err != nil {
+		return err
+	}
+	err = deleteNoLongerNeededRepos(ctx, daoReg)
+	if err != nil {
+		return err
+	}
 	return err
 }
 
+func deleteNoLongerNeededRepos(ctx context.Context, daoReg *dao.DaoRegistry) error {
+	q, err := queue.NewPgQueue(ctx, db.GetUrl())
+	if err != nil {
+		return fmt.Errorf("error getting new task queue: %w", err)
+	}
+	defer q.Close()
+	c := client.NewTaskClient(&q)
+
+	urls := []string{
+		"https://cdn.redhat.com/content/dist/layered/rhel8/x86_64/ansible/2/os/",
+		"https://cdn.redhat.com/content/dist/layered/rhel8/aarch64/ansible/2/os/",
+	}
+	for _, url := range urls {
+		results, _, err := daoReg.RepositoryConfig.List(ctx, "0", api.PaginationData{Limit: 1}, api.FilterData{URL: url, Origin: config.OriginRedHat})
```

Comment:
this works so this is really just a curiosity -- why is org ID "0" here? because org doesn't matter in this case?

Reply:
yep, this is only for RH repos, so the org doesn't matter, but we have to pass something in, passing the RH org id might make more sense.  I'll do that